src/code/z_play.c

```diff
@@ -1,59 +1,1462 @@
+                    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_800BD314.s")
+                    if (globalCtx->envCtx.unk_E1) {} // Necessary to match
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_800BED40.s")
+                    if (globalCtx->envCtx.unk_E1 == 1) {
+                        func_8007672C(gfxCtx, globalCtx->envCtx.unk_E2[0], globalCtx->envCtx.unk_E2[1],
+                                      globalCtx->envCtx.unk_E2[2], globalCtx->envCtx.unk_E2[3], 3);
+                    } else {
+                        // Also necessary to match
+                    }
+                }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_800BEDD8.s")
+                if ((HREG(80) != 10) || (HREG(88) != 0)) {
+                    if (globalCtx->envCtx.unk_E6 != 0) {
+                        func_80076934(globalCtx);
+                    }
+                }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_800BFAE4.s")
+                if ((HREG(80) != 10) || (HREG(93) != 0)) {
+                    DebugDisplay_DrawObjects(globalCtx);
+                }
 
+                if ((R_PAUSE_MENU_MODE == 1) || (D_80161490 == 1)) {
+                    Gfx* sp70 = gfxCtx->overlay.p;
+                    globalCtx->preRenderCtx.unk_10 = gfxCtx->curFrameBuffer;
+                    globalCtx->preRenderCtx.unk_14 = gZBuffer;
+                    func_800C1F20(&globalCtx->preRenderCtx, &sp70);
+                    if (R_PAUSE_MENU_MODE == 1) {
+                        globalCtx->preRenderCtx.unk_18 = gfxCtx->curFrameBuffer;
+                        func_800C20B4(&globalCtx->preRenderCtx, &sp70);
+                        R_PAUSE_MENU_MODE = 2;
+                    } else {
+                        D_80161490 = 2;
+                    }
+                    gfxCtx->overlay.p = sp70;
+                    globalCtx->preRenderCtx.unk_A3 = 2;
+                    SREG(33) |= 1;
+                } else {
+                Gameplay_Draw_DrawOverlay:
+                    if ((HREG(80) != 10) || (HREG(89) != 0)) {
+                        Gameplay_DrawOverlay(globalCtx);
+                    }
+                }
+            }
+        }
+    }
+
+    if (globalCtx->view.unk_124 != 0) {
+        Vec3s sp50;
+        func_800591EC(&sp50, globalCtx->cameraPtrs[globalCtx->activeCamera]);
+        func_800AB944(&globalCtx->view);
+        globalCtx->view.unk_124 = 0;
+        if ((globalCtx->skyboxId != 0) && (globalCtx->skyboxId != 0x1D) && !globalCtx->envCtx.skyDisabled) {
+            func_800B1030(&globalCtx->skyboxCtx, globalCtx->view.eye.x, globalCtx->view.eye.y, globalCtx->view.eye.z);
+        }
+    }
+
+    func_80059EC8(globalCtx->cameraPtrs[globalCtx->activeCamera]);
+
+    Graph_CloseDisps(dispRefs, gfxCtx, "../z_play.c", 4508);
+}
+#else
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/Gameplay_Draw.s")
+#endif
+
+void Gameplay_Main(GlobalContext* globalCtx) {
```

Comment:
I see that this function calls Gameplay_Update.. I guess this confuses me cause update means the same thing as main to me. Would like to hear why you chose this name

Reply:
In the case of game states, the difference is they only have a "main" function instead of separate "update" and "draw" functions. The main function handles both updating and drawing things. For example Gameplay_Main here calls Gameplay_Update then Gameplay_Draw. The same pattern applies to other game states, and it's basically the equivalent of a "main loop" or a "thread entrypoint".
As to why they didn't split "update" and "draw", my guess is that since there is only ever one game state running, there is no need to call all "update" functions then all "draw" functions, like it's done in the case of actors for example.

Reply:
ah ok, that makes sense. thanks